### ISSUE TITLE
PolyPen: Unhandled Exception Caught: Exception caught

### ISSUE THREAD

**Issue by Lewiatan45** [CGCookie/retopoflow#868]
Retopoflow Version: - RetopoFlow: 3.0.0β2

Blender Version/Hash: - Blender: 2.90.0 master 2020-08-31

Platform Version/Distribution:  Windows 10

**Issue:**
PolyPen: Unhandled Exception Caught: Exception caught

**How to Reproduce:**
Unknown, I just wanted to create a new polygon using PolyPen.
{*attach any files needed to replicate the issue*}

Automatically Generated Info
-----------------------------------

```
Environment:

- RetopoFlow: 3.0.0β2
- Blender: 2.90.0 master 2020-08-31
- Platform: Windows, 10, 10.0.18362, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080/PCIe/SSE2, 4.5.0 NVIDIA 456.71, 4.50 NVIDIA
- Timestamp: 2020-10-15 17:28:50.155252
- Undo: insert, insert, move grabbed, select, slide edge loop/strip, insert edge strip, slide edge loop/strip, insert edge strip

Error Hash: 4f26f6d57d9fefb008b1db5c85052259

Trace:

- EXCEPTION (<class 'AssertionError'>): unhandled type of other: None (<class 'NoneType'>)
-     .../common\maths.py
- 000 0186:__sub__() assert False, "unhandled type of other: %s (%s)" % (str(other), str(t))
-     C:\Users\USER\AppData\Roaming\Blender Foundation\Blender\2.90\scripts\addons\RetopoFlow\retopoflow\rftool_polypen\polypen.py
- 001 0542:modal_move() if (xy_updated - xy1).length < self.rfcontext.drawing.scale(options['polypen merge dist']):
-     C:\Users\USER\AppData\Roaming\Blender Foundation\Blender\2.90\scripts\addons\RetopoFlow\retopoflow\rftool.py
- 002 0141:wrapper() ret = fn(*args, **kwargs)
-     .../common\fsm.py
- 003 0054:run() return fn(*args, **kwargs)
```

**Comment by Lewiatan45** [comment]
On a side note, when are you going to release a version of this plugin without horrendous performance issues? (2-3 second input lag for UI buttons, laggy retopo tools by a high poly model that has 3mln polygons)

**Comment by vxlcoder** [comment]
thanks for the report, @Lewiatan45.

I believe this issue is related to issue #867, which _should_ be fixed in the `b280` branch.

We have fixed many of these issues and significantly improved the horrendous performance issue.  We are taking a last pass right now through the issues and code before shipping out our next beta very soon (today/tomorrow).  With high probability, this next beta will be our last before an official release.

Keep in mind that version 3.0 is still in beta, so some issues are to be expected.  We appreciate your work in reporting issues, though!

Thank you for your report and patience on this!

**Comment by Lewiatan45** [comment]
Yes, I noticed a similar issue after I finished writing this one.

I hope you are right because this plugin was in beta for several months and people do lose hope you will be ever able to fix it (check reviews). I just opened my blend file and after dealing with this bad performance for a few minutes I had enough.

I guess the official release will be for 2.90+ versions considering how long getting this released takes. 

PS: I hope you fixed vertex/edge/face selection because it drives me insane sometimes. The user that wrote 1/5 review on your store page explained it well what's wrong with it. 

Have a nice day.
